### ISSUE TITLE
[Research/Advanced]: Automated/Unsupervised Fine-Tuning of Retrieval

### ISSUE THREAD

**Issue by jon-chuang** [run-llama/llama_index#6925]
### Feature Description

- [Augmentation-Adapted Retriever Improves Generalization of Language
Models as Generic Plug-In](https://aclanthology.org/2023.acl-long.136.pdf) shows that one can use a cheap source LM (e.g. Flan-T5-Base - 250M params) to pre-train a retriever (e2e) on a task (MSMARCO) separate to the test-time task. We can look into some of the methodology to see how this could be applied to: 1. fine tuning retrieval on cheap source LM on user's dataset. 2. fine tuning retrieval with the target model on user's dataset (may not be possible as this seems to require access to attention scores on the doc tokens). But it is highly reminiscent of utilizing relevancy scores from reranker to fine-tune.
- [Unsupervised Domain Adaptation](https://arxiv.org/abs/2303.00807)
- [Some other MSFT paper on RAG](https://arxiv.org/pdf/2307.07164.pdf)

### Resources
1. [PyTorch Metric Learning](https://kevinmusgrave.github.io/pytorch-metric-learning/)
